### PR TITLE
Hooks added

### PR DESCRIPTION
**Pull request by aiberry**

### CODE REVIEW

File: src/Task.js [hunk 1]
```diff
@@ -5,7 +5,7 @@ import styles from './Task.module.css';
 export default function Task({ handler, status, name }) {
     return (
         <li
-            onClick={handler}
+            onClick={() => handler()}
```
Comment (pavtaras) [inline src/Task.js]: 1. why not simply `onClick={handler}` ?
2. sorry i didn't notice it previously - `handler` is not great name for the prop. lets call it... `onToggle` may be?

File: src/App.js [hunk 2]
```diff
@@ -1,53 +1,54 @@
...
+                // onKeyUp={searchHandler}
+                onKeyUp={(e) => setQuery(e.target.value)}
+            />
+            <ul>
+                {props.filteredTasks.map((task) => (
+                    <Task
+                        name={task.name}
+                        status={task.status}
+                        key={task.key}
+                        handler={() => taskClicked(task)}
```
Comment (pavtaras) [inline src/App.js]: `onToggle={() => taskClicked(task)}`

File: src/App.js [hunk 3]
```diff
@@ -1,53 +1,54 @@
...
 import Task from './Task.js';
 import { connect } from 'react-redux';
 import addTask from './actions/addTask.js';
 import taskClicked from './actions/taskClicked.js';
 import searchFilter from './actions/searchFilter.js';
 
-class App extends React.Component {
-    taskClicked = (task) => {
-        this.props.onTaskClicked(task);
+function App(props) {
```
Comment (pavtaras) [inline src/App.js]: destructurise props here:

```
function App({ onTaskClicked, onTaskAdded, ... }) {}
```

File: src/App.js [hunk 4]
```diff
@@ -1,53 +1,54 @@
...
+
+    props.onSearchHandler(searchQuery);
+
+    return (
+        <div className={styles.wrap}>
+            <h1>TO-DO List</h1>
+            <input
+                placeholder="Search tasks...."
+                className={styles.typesearchInput}
+                // onKeyUp={searchHandler}
```
Comment (pavtaras) [inline src/App.js]: dont leave commented code

File: src/App.js [hunk 5]
```diff
@@ -1,53 +1,54 @@
...
+                onKeyUp={(e) => setQuery(e.target.value)}
+            />
+            <ul>
+                {props.filteredTasks.map((task) => (
+                    <Task
+                        name={task.name}
```
Comment (pavtaras) [inline src/App.js]: btw, why dont pass `task={task}`?

File: src/App.js (hunk 3 above)
Comment (aiberry) [reply src/App.js]: Fixed

File: src/App.js (hunk 5 above)
Comment (aiberry) [reply src/App.js]: Done

File: src/App.js [hunk 6]
```diff
@@ -1,53 +1,67 @@
...
+                placeholder="Search tasks...."
+                className={styles.typesearchInput}
+                onKeyUp={(e) => setQuery(e.target.value)}
+            />
+            <ul>
+                {filteredTasks.map((task) => (
+                    <Task
+                        task={task}
+                        key={task.key}
+                        onToggle={setClicked}
```
Comment (pavtaras) [inline src/App.js]: why not `onToggle={onTaskClicked}`?

File: src/App.js [hunk 7]
```diff
@@ -1,53 +1,67 @@
...
+    const [newTask, setNewTask] = useState('');
+
+    useEffect(()=>{
+        onTaskClicked(clickedTask);
+    }, [clickedTask])
+    useEffect(()=>{
+        return onTaskAdded(newTask);
+    }, [newTask])
+
+    // const taskAdded = (event) => {
```
Comment (pavtaras) [inline src/App.js]: remove commented code

File: src/App.js [hunk 8]
```diff
@@ -68,3 +82,10 @@ export default connect(
         }
     })
 )(App);
+
+App.propTypes = {
+    onTaskClicked: PropTypes.func,
+    onTaskAdded: PropTypes.func, 
+    onSearchHandler: PropTypes.func, 
+    filteredTasks: PropTypes.array
+};
```
Comment (pavtaras) [inline src/App.js]: add new line

File: src/App.js [hunk 9]
```diff
@@ -1,53 +1,67 @@
...
+    const [searchQuery, setQuery] = useState('');
+    const [clickedTask, setClicked] = useState({});
+    const [newTask, setNewTask] = useState('');
+
+    useEffect(()=>{
+        onTaskClicked(clickedTask);
+    }, [clickedTask])
+    useEffect(()=>{
+        return onTaskAdded(newTask);
+    }, [newTask])
```
Comment (pavtaras) [inline src/App.js]: these 2 useEffect's are redundant

File: src/App.js [hunk 10]
```diff
@@ -1,53 +1,67 @@
...
+                    />
+                ))}
+            </ul>
+            <input
+                placeholder="Add tasks....."
+                className={styles.taskAddInput}
+                onKeyUp={
+                    (e) => {
+                        if (e.keyCode===13) {
+                            setNewTask(e.target.value);
```
Comment (pavtaras) [inline src/App.js]: why not `onTaskAdded(e.target.value)`?

File: src/App.js [hunk 11]
```diff
@@ -1,53 +1,67 @@
...
+    // const taskAdded = (event) => {
+    //     if (event.keyCode === 13) {
+    //         // 13 - Button Enter
+    //         onTaskAdded(event.target.value);
+    //         event.target.value = '';
+    //     }
+    // };
+    useEffect(() => {
+        onSearchHandler(searchQuery);
+    }, [searchQuery])
```
Comment (pavtaras) [inline src/App.js]: actually this useEffect might be also removed but lets leave it as an example of `useEffect`

File: src/App.js (hunk 6 above)
Comment (aiberry) [reply src/App.js]: The only reason for "set..." naming is that this is a setter from useState Hook. Does it really makes sens?

File: src/App.js (hunk 7 above)
Comment (aiberry) [reply src/App.js]: It seems all right in actual commit, smth. strange..

File: src/App.js (hunk 9 above)
Comment (aiberry) [reply src/App.js]: Let's leave them all just as an example of functionality

File: src/App.js (hunk 10 above)
Comment (aiberry) [reply src/App.js]: It seems not a big huge of code. Let's leave it as is

Comment (pavtaras) [reply src/App.js]: it is essential:

1. you make redundant re-renders of app
2. additional code should be tested additionally

**Pull request merged**